### DISCUSSION TITLE
How to auto-backup and auto update?

### DISCUSSION THREAD

**Discussion by PeopleInside** [getgrav/grav#3105] [General]
Hi, let's try to request support on this new GitHub community!
No problem for me as I use GitHub and it's good. This is my first time in trying the GitHub community.

**The question:**

I run 5 or more Grav instance ad want automated update of core and plugins after having a backup.
There is a way I can automate this?

For the backup, if is not possible to ask to the automated system a successfully backup before upgrade, maybe can set an auto backup a week for maximum of 4 backups at month and run a cron for upgrade Grav core and plugins.

Any way to automate the core and plugin update of more instance?
thank you in advance.

Marco

**Comment by rhukster** [comment]
One solution is to use the command line tools and cron to automate. For example you can do a `bin/gpm update -q -n` for quiet and no interaction.  This could be cron'ed to run every day.

A better solution is something we have been working on for a while called **GravGuard**.  This will be an online service that keeps track of all your Grav sites and will allow remote updating via a GravGuard plugin that is installed on each Grav instance.  In the paid plans you will be able to auto-update.  This is not quite ready for primetime, but we hope to have it launched in the new year.

**Reply by PeopleInside** [reply]
Hi, thanks for the reply!
If I have grav installed on **public_html/webiste1/grav** and **public_html/website2/grav** what will be the command to do?
(complete cron)

The command you mentioned will upgrade without backups?

I cannot pay currently, for any service so I must look at the free solution as I have no money to invest and I'm on a personal project (personal websites) just for my use.
